source/common/http/http3/quic_client_connection_factory.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include <string>
+
+#include "envoy/config/listener/v3/quic_config.pb.h"
+#include "envoy/config/typed_config.h"
+#include "envoy/network/connection.h"
+#include "envoy/ssl/context_config.h"
+
+namespace Envoy {
+namespace Http {
+
+// A factory to create EnvoyQuicClientConnection instance for QUIC
+class QuicClientConnectionFactory : public Config::UntypedFactory {
+public:
+  ~QuicClientConnectionFactory() override = default;
+
+  virtual std::unique_ptr<Network::ClientConnection>
+  createQuicNetworkConnection(Network::Address::InstanceConstSharedPtr server_addr,
+                              Network::Address::InstanceConstSharedPtr local_addr,
+                              Network::TransportSocketFactory& transport_socket_factory,
```

Comment:
nits: pass in QuicClientTransportSocketFactory instead.

Reply:
this is a core http3 API and that's a QUIC class in the extension directory so I think leaving as-is makes sense?

Reply:
Ah, I see. That makes sense! Maybe I should move Quic TransportSocketFactory into core code instead. It doesn't depends on QUICHE.

Reply:
I think we should move it all at this point. See my other comments.